keps/2937-resource-transformer/README.md

```diff
@@ -0,0 +1,560 @@
+      Borrowed:  0
+      Name:      cpu
+      Total:     1
+      Borrowed:  0
+      Name:      kueue.x-k8s.io/accelerator-memory
+      Total:     20G
+      Borrowed:  0
+      Name:      memory
+      Total:     97656250Ki
+  Flavors Usage:
+    Name:  default-flavor
+    Resources:
+      Borrowed:         0
+      Name:             cpu
+      Total:            1
+      Borrowed:         0
+      Name:             kueue.x-k8s.io/accelerator-memory
+      Total:            20G
+      Borrowed:         0
+      Name:             memory
+      Total:            97656250Ki
+  Pending Workloads:    0
+  Reserving Workloads:  1
+```
+
+After enough Workloads are admitted that there is queuing,
+then the detailed status message in the `QuotaReserved` condition
+of the Workload will indicate the resource(s) with insufficient quota, but the
+actual transformed resource request is not observable.
+```yaml
+Spec:
+  Active:  true
+  Pod Sets:
+    Count:  1
+    Name:   main
+    Template:
+      Metadata:
+      Spec:
+        Containers:
+          ...
+          Resources:
+            Limits:
+              nvidia.com/mig-1g.5gb:   2
+              nvidia.com/mig-2g.10gb:  1
+            Requests:
+              Cpu:                     1
+              Memory:                  100G
+              nvidia.com/mig-1g.5gb:   2
+              nvidia.com/mig-2g.10gb:  1
+...
+Status:
+  Conditions:
+    Last Transition Time:  2024-09-19T14:59:01Z
+    Message:               couldn't assign flavors to pod set main: insufficient unused quota for kueue.x-k8s.io/accelerator-memory in flavor default-flavor, 20G more needed
+    Observed Generation:   1
+    Reason:                Pending
+    Status:                False
+    Type:                  QuotaReserved
+```
+
+We could improve the observability by extending the Workload Status with a ResourceRequests
+field that would contain a copy of the `TotalRequests` field of the controller-internal `workload.Info`
+struct.  If we did this, we would instead see something like the below for a non-admitted Workload:
+```yaml
+Spec:
+  Active:  true
+  Pod Sets:
+    Count:  1
+    Name:   main
+    Template:
+      Metadata:
+      Spec:
+        Containers:
+          ...
+          Resources:
+            Limits:
+              nvidia.com/mig-1g.5gb:   2
+              nvidia.com/mig-2g.10gb:  1
+            Requests:
+              Cpu:                     1
+              Memory:                  100G
+              nvidia.com/mig-1g.5gb:   2
+              nvidia.com/mig-2g.10gb:  1
+...
+Status:
+  Conditions:
+    Last Transition Time:  2024-09-19T14:59:01Z
+    Message:               couldn't assign flavors to pod set main: insufficient unused quota for kueue.x-k8s.io/accelerator-memory in flavor default-flavor, 20G more needed
+    Observed Generation:   1
+    Reason:                Pending
+    Status:                False
+    Type:                  QuotaReserved
+  ResourceRequests:
```

Comment:
Yes, some form of a field like this will be important, but if we have multiple PodSets we may need to keep the structure per PodSet.

Alternatively, we can wait with adding the field, but start with "Alpha" phase, and adding some form of the field before Beta. We may actually prefer to go via Alpha regardless, WDYT?

Reply:
I have no strong opinion about going Alpha vs. Beta as the first target.  I guess if Alpha, then I should add a feature gate to the current prototype implementation PR?

Reply:
Going via Alpha would be nice, so that we can easily adjust the API before Beta. Please add a feature gate, maybe `ConfigurableResourceTransformations` per KEP title.

Reply:
Added feature gate to KEP.yaml